WorldCat Discovery Service.js

```diff
@@ -0,0 +1,287 @@
+{
+	"translatorID": "fd8dc5f6-a6dd-42b2-948f-600f5da844ea",
+	"label": "WorldCat Discovery Service",
+	"creator": "Sebastian Karcher",
+	"target": "^https?://[^/]+\\.worldcat\\.org",
+	"minVersion": "3.0.9",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsbv",
+	"lastUpdated": "2015-06-07 17:32:35"
+}
+
+function detectWeb(doc, url) {
+	
+	var results = getSearchResults(doc);
+	if(results.length) {
+		return "multiple";
+	}
+	
+
+	//single result
+	// generate item and return type
+	var co = getFirstContextObj(doc);
+	if(!co) return false;
+
+	return generateItem(doc, co).itemType;
+}
+
+
+/**
+ * Gets Zotero item from a WorldCat icon src
+ */
+function getZoteroType(iconSrc) {
+	// only specify types not specified in COinS
+	if (iconSrc.indexOf("icon-rec") != -1) {
+		return "audioRecording";
+	}
+	if (iconSrc.indexOf("icon-com") != -1) {
+		return "computerProgram";
+	}
+	if (iconSrc.indexOf("icon-map") != -1) {
+		return "map";
+	}
+	return false;
+}
+
+/**
+ * Generates a Zotero item from a single item WorldCat page,
+ * or the first item on a multiple item page
+ */
+function generateItem(doc, co) {
+	var item = new Zotero.Item();
+	ZU.parseContextObject(co, item);
+	// item types not covered by COinS will still need to be covered. Leaving the corresponding code from open Worldcat
+	// here as a reminder.
+
+/*
+	var type = ZU.xpathText(doc,
+		'//img[@class="icn"][contains(@src, "icon-")][1]/@src');
+	if (type) {
+		type = getZoteroType(type);
+		if (type) item.itemType = type;
+	} */
+	
+	return item;
+}
+
+function getSearchResults(doc) {
+	var results = ZU.xpath(doc, '//ol[@class="results"]/li[contains(@id, "record")]')
+	return results;
+}
+
+function getTitleNode(searchResult) {
+	return ZU.xpath(searchResult, './div[@class="title"]/a')[0];
+}
+
+function getFirstContextObj(doc) {
+	return ZU.xpathText(doc, '//span[@class="Z3988"][1]/@title');
+}
+
+
+
+/**
+ * Given an item URL, extract OCLC ID
+ */
+function extractOCLCID(url) {
+	var id = url.match(/\/oclc\/([^?]+)/);
+	if(!id) return false;
+	return id[1];
+}
+
+
+/**
+ * Given an item URL, extract database ID
+ */
+function extractDatabaseID(url) {
+	var id = url.match(/databaseList=[^&#]+/);
```

Comment:
Also, just for completeness and to make sure we don't mess up later, `decodeURIComponent` here then encode it back when building url